src/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/RuntimeTypeInfo.TypeComponentsCache.cs

```diff
@@ -52,7 +51,7 @@ public TypeComponentsCache(RuntimeTypeInfo type)
 
                 object unifierAsObject = cacheArray[index];
                 Debug.Assert(unifierAsObject is PerNameQueryCache<M>);
-                PerNameQueryCache<M> unifier = Unsafe.As<PerNameQueryCache<M>>(unifierAsObject);
```

Comment:
I am not sure why Unsafe.As was used here. If there was an issue with the blob, then wouldn't an explicit cast be better to detect?